tools/build/mingw_toolchain/file_collector.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <vector>
+
+// Keeps track of tempoarary files and processes @ params files
+class FileCollector {
+ public:
+  FileCollector(const std::string& baseName);
+  ~FileCollector();
+
+  enum ParamsType {
+    GCC, AR
```

Comment:
Please can we have a little bit of documentation here to describe the differences between the two of these?

Reply:
Done.